tasks/eva_2950/split_rs_with_inconsistent_ss.py

```diff
@@ -0,0 +1,307 @@
+
+    return normalize(*leftalign(chrom, pos, ref, alt, fa), left_only=True)
+
+
+def add_contex_base(fa, chrom, pos, ref, alt):
+    # Add the context base if it is not there already
+    if ref == '':
+        ref = fa[chrom][pos - 2].upper()  # -1 because pos is 1-based and -1 because we want the base before
+        pos -= 1
+        alt = ref + alt
+    if alt == '':
+        alt = fa[chrom][pos - 2].upper()
+        pos -= 1
+        ref = alt + ref
+    return pos, ref, alt
+
+def ast_parse_with_datetime(astr):
+    """
+    Parse a string representing a python data structure into a python data structure
+    Based on https://stackoverflow.com/questions/4235606/way-to-use-ast-literal-eval-to-convert-string-into-a-datetime
+    """
+    try:
+        tree = ast.parse(astr)
+    except SyntaxError:
+        raise ValueError(astr)
+    for node in ast.walk(tree):
+        if isinstance(node, (ast.Module, ast.Expr, ast.Dict, ast.Str, ast.List, ast.Constant,
+                            ast.Attribute, ast.Num, ast.Name, ast.Load, ast.Tuple)):
+            continue
+        if isinstance(node, ast.Call) and isinstance(node.func, ast.Attribute) and node.func.attr == 'datetime':
+            continue
+        raise ValueError(astr)
+    return eval(astr)
+
+
+def parse_eva2850_diagnositc_log(log_file):
+    rsid = None
+    with open(log_file) as open_file:
+        for line in open_file:
+            line = line.strip()
+            if rsid:
+                # We are parsing the line just after finding the RS id
+                list_of_ssids = ast_parse_with_datetime(line.strip()[11:])
+                yield rsid, list_of_ssids
+                rsid = None
+            if 'Not all original SS has same info' in line:
+                sp_line = line.split()
+                rsid = int(sp_line[4].strip(','))
+
+        if rsid:
+            list_of_ssids = ast_parse_with_datetime(line.strip()[11:])
+            yield rsid, list_of_ssids
+
+
+genome_cache = {}
+
+
+def get_genome_object(ref_genome_directory, genome_accession):
+    if genome_accession not in genome_cache:
+        path_to_search = os.path.join(ref_genome_directory, '*', genome_accession, genome_accession + '.fa')
+        genome_paths = glob.glob(path_to_search)
+        if len(genome_paths) == 0:
+            raise ValueError(f'Cannot locate genome for {genome_accession} in {ref_genome_directory}')
+        genome_path = genome_paths[0]
+        genome_cache[genome_accession] = Fasta(genome_path, as_raw=True, read_ahead=40000)
+    return genome_cache[genome_accession]
+
+
+def variant_type(ref, alt):
+    if len(ref) == len(alt) == 1:
+        return 'SNP'
+    if len(ref) > 1 and len(alt) > 1:
+        return 'MNP'
+    if len(ref) > 1:
+        return 'DEL'
+    return 'INS'
+
+
+def shelve_submitted_variant_entities(mongo_handle, submitted_variant_ids):
+    output_collection = 'eva2950_dbsnpSubmittedVariantEntity'
+    query_match = {'_id': {'$in': submitted_variant_ids}}
```

Comment:
Do we know how many ids are here? if there are a lot, the query below may fail...

Reply:
There are ~ 100K but might be more so I change the shelving method to do it in batch of 1000.